src/App.js

```diff
@@ -1,11 +1,18 @@
 import React from "react";
 import Header from "./components/Header.js";
+import CharacterList from "./components/CharacterList.js";
+import { Route } from "react-router-dom";
+import SearchForm from "./components/SearchForm";
+
 
 
 export default function App() {
   return (
     <main>
-      <Header />
+      <Header/>
+      <Route exact path="/" component={WelcomePage} />
+      <Route path="/characters" component={CharacterList} />
```

Comment:
Given that you're passing the functions inside your characterlist component, you want to go with a `render` instead.